org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/codeactions/LSPCodeActionMarkerResolution.java

```diff
@@ -147,31 +161,74 @@ private void checkMarkerResoultion(IMarker marker) throws IOException, CoreExcep
+				if (editor instanceof ITextEditor) {
+					try {
+						Diagnostic diagnostic = (Diagnostic)marker.getAttribute(LSPDiagnosticsToMarkers.LSP_DIAGNOSTIC);
+						final ITextViewer textViewer = ((ITextEditor) editor).getAdapter(ITextViewer.class);
+						if (textViewer != null) {
+							for (CompletableFuture<LanguageServer> lsf : languageServerFutures) {
+								marker.setAttribute(LSP_REMEDIATION, COMPUTING);
+								CodeActionContext context = new CodeActionContext(Collections.singletonList(diagnostic));
+								CodeActionParams params = new CodeActionParams();
+								params.setContext(context);
+								params.setTextDocument(new TextDocumentIdentifier(LSPEclipseUtils.toUri(marker.getResource()).toString()));
+								params.setRange(diagnostic.getRange());
+								CompletableFuture<List<Either<Command, CodeAction>>> codeAction = lsf
+										.thenComposeAsync(ls -> ls.getTextDocumentService().codeAction(params));
+								futures.add(codeAction);
+								codeAction.thenAcceptAsync(actions -> {
+									try {
+										marker.setAttribute(LSP_REMEDIATION, actions);
+										PlatformUI.getWorkbench().getDisplay().asyncExec(() -> reinvokeQuickfixProposalsIfNecessary(textViewer));
+									} catch (CoreException e) {
+										LanguageServerPlugin.logError(e);
+									}
+								});
+							}
 						}
-					});
+					} catch (Exception e) {
+						LanguageServerPlugin.logError(e);
+					}
 				}
+
 				// wait a bit to avoid showing too much "Computing" without looking like a freeze
 				CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()])).get(300, TimeUnit.MILLISECONDS);
 			}
 		}
 	}
 
+	private void reinvokeQuickfixProposalsIfNecessary(ITextViewer textViewer) {
+		try {
+			// Quick assist proposals popup case
+			if (textViewer instanceof ISourceViewerExtension3) {
+				IQuickAssistAssistant quickAssistant = ((ISourceViewerExtension3)textViewer).getQuickAssistAssistant();
+					Field f = QuickAssistAssistant.class.getDeclaredField("fQuickAssistAssistantImpl"); //$NON-NLS-1$
+					f.setAccessible(true);
+					ContentAssistant ca = (ContentAssistant) f.get(quickAssistant);
+					Method m = ContentAssistant.class.getDeclaredMethod("isProposalPopupActive"); //$NON-NLS-1$
+					m.setAccessible(true);
+					boolean isProposalPopupActive = (Boolean) m.invoke(ca);
+					if (isProposalPopupActive) {
+						quickAssistant.showPossibleQuickAssists();
+					}
+			}
+			// Hover case
+			if (textViewer instanceof ITextViewerExtension2) {
+				ITextHover hover = ((ITextViewerExtension2) textViewer).getCurrentTextHover();
+				boolean hoverShowing = hover != null;
+				if (hoverShowing) {
+					Field f = TextViewer.class.getDeclaredField("fTextHoverManager"); //$NON-NLS-1$
+					f.setAccessible(true);
+					AbstractInformationControlManager manager = (AbstractInformationControlManager) f.get(textViewer);
+					manager.showInformation();
```

Comment:
Would re-calling the Hover action against the editor result in a refresh? Something like `editor.getAction(ITextEditorActionConstants.SHOW_INFORMATION)`

Reply:
I've tried `editor.getAction(ITextEditorActionConstants.SHOW_INFORMATION)` previously and it does bring the popup  correctly but there are 2 caches:
- Needs the editor cursor to be located where the popup to appear
- The popup does not go away until a click outside the shell is received. On the contrary, the regular hover goes away when the mouse is moved off the hover shell.